erc20-bridge-token/test/BridgeToken.js

```diff
@@ -298,4 +322,118 @@ describe('BridgeToken', () => {
       .be
       .revertedWith(`AccessControl: account ${user.address.toLowerCase()} is missing role ${ADMIN_ROLE}`);
   })
+
+  describe("Whitelist", function() {
+    let tokenInfo;
+    let testProofId = 1;
+    const recipient = "testrecipient.near";
+
+    beforeEach(async function() {
+      tokenInfo = await createEmptyToken(
+        nearTokenId,
+        BridgeTokenFactory,
+        BridgeTokenInstance
+      );
+      setMetadata(nearTokenId, tokenInfo.tokenProxyAddress);
+
+      const lockResultProof = require('./proof_template.json');
+      lockResultProof.outcome_proof.outcome.status.SuccessValue = serialize(
+        SCHEMA,
+        "LockResult",
+        {
+          prefix: RESULT_PREFIX_LOCK,
+          token: nearTokenId,
+          amount: 100,
+          recipient: ethers.utils.arrayify(adminAccount.address),
+        }
+      ).toString("base64");
+      const source = "123456789FGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+      lockResultProof.outcome_proof.outcome.receipt_ids[0] = source[
+        testProofId
+      ].repeat(44);
+      testProofId += 1;
+      await BridgeTokenFactory.deposit(
+        borshifyOutcomeProof(lockResultProof),
+        1090
+      );
+    });
+
+    it("Test account in whitelist", async function() {
+      await BridgeTokenFactory.setTokenWhitelistMode(nearTokenId, 3);
+      await BridgeTokenFactory.addAccountToWhitelist(
+        nearTokenId,
+        adminAccount.address
+      );
```

Comment:
IMO it's better to use a non-admin account to test the behavior and also perform appropriate further operations (deposit, withdraw) from the user account too.

Reply:
11eea06736ce1a2c09d4118f00d10b4a75b72383